addons/validate_mission/fn_run.sqf

```diff
@@ -0,0 +1,143 @@
+/*
+ * See https://github.com/freghar/arma-additions/blob/master/addons/editor_extensions/validate_mission/README.md
+ */
+
+private _check_game_type = {
+    private _type = "Multiplayer" get3DENMissionAttribute "GameType";
+    if (_type in ["Coop","DM"]) then {
+        ["Game Type", true]
+    } else {
+        ["Game Type", false, [], [
+            format ["Unexpected game type %1.", _type],
+            "This may be okay for a custom mission, but none of the default modules",
+            "use this type. If you're making a normal mission, check that a module",
+            "composition was placed."
+        ]]
+    };
+};
+
+private _check_respawn_type = {
+    private _gametype = "Multiplayer" get3DENMissionAttribute "GameType";
+    private _respawn = "Multiplayer" get3DENMissionAttribute "Respawn";
+    if (!(_gametype in ["Coop","DM"])) exitWith {
+        ["Respawn Type", true]  /* unknown, pass */
+    };
+    /* 3 = custom position, 1 = spectator */
+    if (
+        (_gametype == "Coop" && _respawn in [3,1]) ||
+        (_gametype == "DM" && _respawn == 1)
+        ) then {
+        ["Respawn Type", true]
+    } else {
+        ["Respawn Type", false, [], [
+            "Invalid respawn settings for Coop/PvP.",
+            "Make sure the correct module composition was placed and respawn settings",
+            "were not edited."
+        ]]
+    };
+};
+
+private _check_respawn_marker_invis = {
+    private _markers = all3DENEntities select 5;
+    private _respawns = _markers select { _x select [0,7] == "respawn" };
+    private _failed = [];
+    {
+        private _alpha = _x get3DENAttribute "alpha" select 0;
+        if (_alpha > 0) then {
+            _failed pushBack _x;
+        };
+    } forEach _respawns;
+
+    if (_failed isEqualTo []) then {
+        ["Respawn Markers Invisible", true]
+    } else {
+        ["Respawn Markers Invisible", false, _failed, [
+            "Some respawn markers are visible.",
+            "Double click them in the left panel (or on the map screen) and set their",
+            "'Alpha' value all the way to 0%."
+        ]]
+    };
+};
+
+private _check_respawn_marker_proximity = {
```

Comment:
Maybe recommend/check for a "move on JIP" module to determine whether to pass/fail.

Respawn marker far away from players + move on JIP -> Probably intentional, pass
Respawn far away from players + no move on jip -> Maybe an accident, maybe MM isn't aware that JIPs don't spawn on respawn so fail + advise on a respawn marker move or to create a move on JIP module.

Reply:
Yeah, good point. I could also check the proximity of "Teleport on JIP" to the respawn marker, but maybe that's not worth it.